apps/web/src/components/AccountSelectorModal/AccountSelectorPopover.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  Box,
+  Button,
+  IconButton,
+  Popover,
+  PopoverBody,
+  PopoverContent,
+  PopoverTrigger,
+  Portal,
+  Text,
+} from "@chakra-ui/react";
+import { useDynamicDisclosureContext } from "@umami/components";
+import {
+  type Account,
+  type LedgerAccount,
+  type SecretKeyAccount,
+  type SocialAccount,
+} from "@umami/core";
+import { useSelectedNetwork } from "@umami/state";
+
+import { AccountInfoModal } from "./AccountInfoModal";
+import { RemoveAccountModal } from "./RemoveAccountModal";
+import { RenameAccountPage } from "./RenameAccountModal";
+import {
+  EditIcon,
+  ExternalLinkIcon,
+  SearchIcon,
+  ThreeDotsIcon,
+  TrashIcon,
+} from "../../assets/icons";
+import { useColor } from "../../styles/useColor";
+
+type AccountSelectorPopoverProps = {
+  account: Account;
+};
+
+export const AccountSelectorPopover = ({ account }: AccountSelectorPopoverProps) => {
+  const { openWith } = useDynamicDisclosureContext();
+  const color = useColor();
+  const network = useSelectedNetwork();
+
+  return (
+    <Popover variant="dropdown">
+      <PopoverTrigger>
+        <IconButton
+          alignSelf="flex-end"
+          color={color("500")}
+          background="transparent"
+          aria-label="Account actions"
+          icon={<ThreeDotsIcon />}
+          onClick={event => event.stopPropagation()}
+          size="xs"
+        />
+      </PopoverTrigger>
+      <Portal>
+        <PopoverContent maxWidth="204px">
+          <PopoverBody color={color("400")}>
+            <Box>
+              <Button
+                onClick={async e => {
+                  e.stopPropagation();
+                  await openWith(<AccountInfoModal account={account} />);
+                }}
+                variant="dropdownOption"
+              >
+                <SearchIcon />
+                <Text color={color("900")} fontWeight="600">
+                  Account Info
+                </Text>
+              </Button>
+              <Button
+                onClick={async e => {
```

Comment:
nit: there is no need for async await.

you can just `return openWith(...)`
(and in other places too)